drivers/android/process.rs

```diff
@@ -261,24 +261,24 @@ unsafe impl Sync for Process {}
 
 impl Process {
     fn new(ctx: Ref<Context>) -> Result<Ref<Self>> {
-        Ref::try_new_and_init(
-            Self {
-                ctx,
-                task: Task::current().group_leader().clone(),
-                // SAFETY: `inner` is initialised in the call to `mutex_init` below.
-                inner: unsafe { Mutex::new(ProcessInner::new()) },
-                // SAFETY: `node_refs` is initialised in the call to `mutex_init` below.
-                node_refs: unsafe { Mutex::new(ProcessNodeRefs::new()) },
-            },
-            |mut process| {
-                // SAFETY: `inner` is pinned when `Process` is.
-                let pinned = unsafe { process.as_mut().map_unchecked_mut(|p| &mut p.inner) };
-                kernel::mutex_init!(pinned, "Process::inner");
-                // SAFETY: `node_refs` is pinned when `Process` is.
-                let pinned = unsafe { process.as_mut().map_unchecked_mut(|p| &mut p.node_refs) };
-                kernel::mutex_init!(pinned, "Process::node_refs");
-            },
-        )
+        let mut process = Pin::from(UniqueRef::try_new(Self {
+            ctx,
+            task: Task::current().group_leader().clone(),
+            // SAFETY: `inner` is initialised in the call to `mutex_init` below.
+            inner: unsafe { Mutex::new(ProcessInner::new()) },
+            // SAFETY: `node_refs` is initialised in the call to `mutex_init` below.
+            node_refs: unsafe { Mutex::new(ProcessNodeRefs::new()) },
+        })?);
+
+        // SAFETY: `inner` is pinned when `Process` is.
+        let pinned = unsafe { process.as_mut().map_unchecked_mut(|p| &mut p.inner) };
+        kernel::mutex_init!(pinned, "Process::inner");
```

Comment:
Is it unsafe to init a mutex twice?

Reply:
It is ok to re-init a mutex that isn't locked. It's a problem if it's locked.

Perhaps we should add that the mutex should be inited before use, and never again.

Do you have another concern in mind?

Reply:
> It is ok to re-init a mutex that isn't locked.

:+1:

> It's a problem if it's locked.

That shouldn't be possible with `mutex_init!()` as it requires a mutable reference, right?

> Do you have another concern in mind?

No, I was just worried that it would be possible to cause UB using the safe `mutex_init!()`.

Reply:
> > It's a problem if it's locked.
> 
> That shouldn't be possible with `mutex_init!()` as it requires a mutable reference, right?

It's possible to do the following: init mutex, lock, forget guard, init again. This is in theory not ok (because we're initing a mutex that is locked), but happens to work because the forgotten guard can never unlock the mutex because it was forgotten.